airqo-platform/AirQo-api
# Spark jobs with Kafka on kubernetes

# Chore

**_WHAT DOES THIS PR DO?_**

Sets up and explores the usage of Kafka to manage topics and Spark to manage streaming while harnessing the resources of Kubernetes. Also introduces the concepts/setups for "Infrastructure as Code".

- [x] Changes the file organisation for the infrastructure code 

- [ ] Adds infrastructure setup code (Terraform) 

- [x] Sets up Kafka on Kubernetes using Strimzi 

- [ ] Introduces ENV separation during infra setup and configuration.

**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**
N/A

**_WHAT IS THE LINK TO THE PR BRANCH_**
https://github.com/airqo-platform/AirQo-api/tree/chore-kafka-spark

**_HOW DO I TEST OUT THIS PR?_**
cd AirQo-api/infra

> for setup (Terraform)

`cd setup`
`cd {worker directory}`
`terraform init`
`terraform apply`

> for configuration (Ansible)

`cd config`
Ensure that your SSH key is added to the respective VMS you are configuring (`cat ~/.ssh/id_rsa.pub`)
` ansible-playbook -i hosts/hosts plays/initial.yml`
  `ansible-playbook -i hosts/hosts plays/kube-dependencies.yml`
` ansible-playbook -i hosts/hosts plays/master.yml`
`ansible-playbook -i hosts/hosts plays/workers.yml`
`ssh username@{MASTER_NODE_IP}`


**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 
N/A

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**
N/A

**_ARE THERE ANY RELATED PRs?_**
N/A
